sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -235,8 +235,13 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
           .getOrElse(createJoinWithoutHint())
 
       case j @ ExtractSingleColumnNullAwareAntiJoin(leftKeys, rightKeys) =>
-        Seq(joins.BroadcastHashJoinExec(leftKeys, rightKeys, LeftAnti, BuildRight,
-          None, planLater(j.left), planLater(j.right), isNullAwareAntiJoin = true))
+        if (!canBroadcastBySize(j.right, conf) && conf.adaptiveExecutionEnabled) {
```

Comment:
Just a high level question - if we are worried about broadcast join OOM on driver side here, shouldn't we be also worried about shuffled hash join OOM on task side as well?

Reply:
the HashedRelation built in ShuffledHashedJoinExec in partitioned, it should be less data, so the driver OOM issue should be a major concern.

Reply:
Sorry I don't get why we don't worry about shuffled hash join OOM if sub-query is way too big/skewed in certain keys. In general, shuffled hash join OOM is a normal issue if people choose shuffled hash join blindly. Could we add a check to make sure build size is not too big to be shuffled hash join?

e.g.

```
if (!canBroadcastBySize(j.right, conf) && canBuildLocalHashMapBySize(j.right, conf) && conf.adaptiveExecutionEnabled) {
  ...
}
```

Reply:
> Sorry I don't get why we don't worry about shuffled hash join OOM if sub-query is way too big/skewed in certain keys. In general, shuffled hash join OOM is a normal issue if people choose shuffled hash join blindly. Could we add a check to make sure build size is not too big to be shuffled hash join?
> 
> e.g.
> 
> ```
> if (!canBroadcastBySize(j.right, conf) && canBuildLocalHashMapBySize(j.right, conf) && conf.adaptiveExecutionEnabled) {
>   ...
> }
> ```

Do you have any contact that I can offline communicate with you maybe, maybe you can reach me with my WeChat "18518298234"